include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_two_stage_xdl_cshuffle.hpp

```diff
@@ -638,14 +675,18 @@ struct DeviceGroupedConvBwdWeightTwoStage_Xdl_CShuffle
 
         std::size_t GetWorkspaceETensorSizeBytes() const
         {
-            return sizeof(AccDataType) * ce_grid_desc_m_n_.GetElementSpaceSize() * Conv_G_;
+            // Align to 128B
+            return math::integer_divide_ceil(sizeof(AccDataType) *
+                                                 ce_grid_desc_m_n_.GetElementSpaceSize() * Conv_G_,
+                                             128) *
+                   128;
         }
 
         std::size_t GetWorkspaceSizeBytes() const
         {
             // Transpose require workspace for A and B
-            if constexpr(is_NGCHW_GKYXC_NGKHW<InLayout, WeiLayout, OutLayout>() ||
-                         is_NGCDHW_GKZYXC_NGKDHW<InLayout, WeiLayout, OutLayout>())
+            if constexpr(is_NGCHW_NGKHW<InLayout, WeiLayout, OutLayout>() ||
+                         is_NGCDHW_NGKDHW<InLayout, WeiLayout, OutLayout>())
```

Comment:
Isn't this check too permissive? With layout: `NGCHW_GKCYX_NGKHW` you would fall into else and I assume still transpose of A & B is needed ?

Reply:
We still need transpose for NGCHW_GKCYX_NGKHW.

Reply:
OK so correct me if I get the logic wrong: 
1. We transpose A & B only if needed like in condition above: `NGCHW_NGKHW...`
2. We always ( ?? is that true ? ) transpose E ??
3.  We always need e workspace due to transposition or two stage approach ?

It would be great to have those assumptions written down in comments eg here in this function.